families/sawtooth_validator_registry/validator_registry/src/validator_registry_tp_handler.rs

```diff
@@ -0,0 +1,269 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -----------------------------------------------------------------------------
+ */
+
+extern crate base64;
+
+use crypto::digest::Digest;
+use crypto::sha2::Sha256;
+use sawtooth_sdk::messages::processor::TpProcessRequest;
+use sawtooth_sdk::processor::handler::ApplyError;
+use sawtooth_sdk::processor::handler::TransactionContext;
+use sawtooth_sdk::processor::handler::TransactionHandler;
+use std::collections::HashMap;
+use std::convert::From;
+
+use protobuf::RepeatedField;
+use protos::validator_registry::{
+    ValidatorInfo, ValidatorMap, ValidatorMap_Entry, ValidatorRegistryPayload,
+};
+
+use validator_registry_tp_verifier::verify_signup_info;
+
+const VALIDATOR_MAP_STR: &str = "validator_map";
+const VALIDATOR_REGISTRY_STR: &str = "validator_registry";
+
+pub fn parse_from<T>(data: &[u8]) -> Result<T, ApplyError>
+where
+    T: protobuf::Message,
+{
+    protobuf::parse_from_bytes(&data).map_err(|err| {
+        warn!(
+            "Invalid error: Failed to parse ValidatorRegistryTransaction: {:?}",
+            err
+        );
+        ApplyError::InternalError(format!(
+            "Failed to unmarshal ValidatorRegistryTransaction: {:?}",
+            err
+        ))
+    })
+}
+
+fn set_state<T>(context: &mut TransactionContext, address: &str, data: T) -> Result<(), ApplyError>
+where
+    T: protobuf::Message,
+{
+    let bytes = protobuf::Message::write_to_bytes(&data)
+        .map_err(|err| ApplyError::InternalError(format!("Failed to serialize: {:?}", err)))?;
+    let mut map: HashMap<String, Vec<u8>> = HashMap::new();
+    map.insert(address.to_string(), bytes);
+    context.set_state(map).map_err(|_| {
+        warn!("Failed to save value at address {}", address);
+        ApplyError::InternalError("Unable to save to state".to_string())
+    })?;
+    Ok(())
+}
+
+pub fn get_state(
+    context: &mut TransactionContext,
+    address: &str,
+) -> Result<Option<Vec<u8>>, ApplyError> {
+    context.get_state(vec![address.to_string()]).map_err(|err| {
+        warn!("Internal Error: Failed to load state: {:?}", err);
+        ApplyError::InternalError(format!("Failed to load state: {:?}", err))
+    })
+}
+
+pub struct ValidatorRegistryTransactionHandler {
+    family_name: String,
+    family_versions: Vec<String>,
+    namespaces: Vec<String>,
+}
+
+impl ValidatorRegistryTransactionHandler {
+    pub fn new() -> ValidatorRegistryTransactionHandler {
+        ValidatorRegistryTransactionHandler {
+            family_name: String::from(VALIDATOR_REGISTRY_STR),
+            family_versions: vec![String::from("2.0")],
```

Comment:
Can the version also be declared as a string constant similar to VR TP name?

Reply:
Maybe